impeller/renderer/backend/vulkan/allocator_vk.cc

```diff
@@ -234,7 +333,14 @@ class AllocatedTextureSourceVK final : public TextureSourceVK {
 
     alloc_nfo.usage = ToVMAMemoryUsage();
     alloc_nfo.preferredFlags = ToVKMemoryPropertyFlags(desc.storage_mode);
-    alloc_nfo.flags = ToVmaAllocationCreateFlags(desc.storage_mode, true);
+
+    auto image_upload = desc.usage_hint == UsageHint::kImageUpload;
+    alloc_nfo.flags = ToVmaAllocationCreateFlags(
+        desc.storage_mode, /*is_texture=*/true,
+        image_upload ? 0u : desc.GetByteSizeOfBaseMipLevel());
```

Comment:
Sort of hacky, but if we want to allocate in the specific pool ... we can't ask for dedicated memory. Makes sense

Reply:
This may be a bit more clear if you collapse it to one branch instead of 2 identical branches next to each other.